src/pages/storage/actions/snapshots/VolumeAddSnapshotBtn.tsx

```diff
@@ -37,9 +39,11 @@ const VolumeAddSnapshotBtn: FC<Props> = ({
           title={
             isDisabled
               ? `Snapshot creation is blocked for project ${volume.project}`
-              : "Add Snapshot"
+              : !canManageStorageVolumeSnapshots(volume)
+                ? "You do not have permission to create or delete snapshots of this volume."
+                : "Add Snapshot"
```

Comment:
Probably better to extract this into a function with early exits as well. Then the disabled check below can use that same function. See how `DeleteStoragePoolBtn` is doing it for an example.